### ISSUE TITLE
Nice way to calculate measures on train

### ISSUE THREAD

**Issue by ja-thomas** [mlr-org/mlr#1067]
It would be nice to have a a more convenient way to specify measures that should be calculated on the training set in resample. Compare issue #1044. 

We have a PR for documentation of the problem.

It would be cool if we can do something like

```
resample(..., measures = list(mmce, onTrain(mmce)))
```

and then call the same code unexposed to the user.

**Comment by MasonGallo** [comment]
Tagging on here, any reason why not to have `makeResampleDesc` have predict = 'both' as the default other than memory? That way @ja-thomas proposed `onTrain` wouldn't have to alter this behind the scenes, which would be confusing IMO.

I almost always want to see both test/train in my work. If users don't want to hold both sets, they can switch it off.

**Comment by ja-thomas** [comment]
I like that idea more, the question is if there are situations where I do want some measures to be evaluated on test + train, some only on test and some only on train. I not sure if there are use-cases for stuff like that. If nobody can think of something I would say @MasonGallo s suggestion is better

**Comment by berndbischl** [comment]
> Tagging on here, any reason why not to have makeResampleDesc have predict = 'both' as the default other than memory? That way @ja-thomas proposed onTrain wouldn't have to alter this behind the scenes, which would be confusing IMO.
> 
> I almost always want to see both test/train in my work. If users don't want to hold both sets, they can switch it off.

the reason is memory and time. (computing the predictions is not free....)
and many people ONLY want them on the test set. i dont want to alter the default here, as this would be a behavior change of the package. one where i dont even see this as an improvement.

**Comment by catastrophic-failure** [comment]
I can't see any use for predictions on train set other than debugging in my applications.

**Comment by berndbischl** [comment]
> I can't see any use for predictions on train set other than debugging in my applications.

i can understand mason as well. and i sometimes like to look at training performance too.
but rarely. the point is: it different in different scenarios. and we already have a standard here. 

so IMHO we should keep it like it is, but think about how to make it easier for guys like mason to ask mlr for what they want - in this case training performance

which is the exact raison d'etre of this thread ;-)
